src/main/kotlin/app/revanced/patcher/apk/ApkBundle.kt

```diff
@@ -0,0 +1,156 @@
+import com.reandroid.arsc.util.FrameworkTable
+import com.reandroid.arsc.value.ResConfig
+import com.reandroid.common.TableEntryStore
+import java.io.File
+
+/**
+ * An [Apk] file of type [Apk.Split].
+ *
+ * @param files A list of apk files to load.
+ */
+class ApkBundle(files: List<File>) {
+    /**
+     * The [Apk.Base] of this [ApkBundle].
+     */
+    val base: Apk.Base
+
+    /**
+     * A map containing all the [Apk.Split]s in this bundle associated by their configuration.
+     */
+    val splits: Map<String, Apk.Split>?
+
+    private fun ApkModule.isFeatureModule() = androidManifestBlock.manifestElement.let {
+        it.searchAttributeByName("isFeatureSplit")?.valueAsBoolean == true || it.searchAttributeByName("configForSplit") != null
+    }
+
+    init {
+        var baseApk: Apk.Base? = null
+        val splitList = mutableListOf<Apk.Split>()
+
+        files.forEach {
+            val module = ApkModule.loadApkFile(it)
+            when {
+                module.isBaseModule -> {
+                    if (baseApk != null) {
+                        throw IllegalArgumentException("Cannot have more than one base apk")
+                    }
+                    baseApk = Apk.Base(module)
+                }
+
+                !module.isFeatureModule() -> {
+                    val config = module.split.removePrefix("config.")
+
+                    splitList.add(
+                        when {
+                            config.length == 2 -> Apk.Split.Language(config, module)
+                            Apk.Split.Library.architectures.contains(config) -> Apk.Split.Library(config, module)
+                            ResConfig.Density.valueOf(config) != null -> Apk.Split.Asset(config, module)
+                            else -> throw IllegalArgumentException("Unknown split: $config")
+                        }
+                    )
+                }
+            }
+        }
+
+        splits = splitList.takeIf { it.size > 0 }?.let { splitList.associateBy { it.config } }
+        base = baseApk ?: throw IllegalArgumentException("Base apk not found")
+    }
+
+    /**
+     * A [Sequence] yielding all [Apk]s in this [ApkBundle].
+     */
+    val all = sequence {
+        yield(base)
+        splits?.values?.let {
+            yieldAll(it)
+        }
+    }
+
+    /**
+     * Write all the [Apk]s inside the bundle to a folder.
+     *
+     * @param options The [PatcherOptions] of the [Patcher].
+     * @param folder The folder to write the [Apk]s to.
+     * @return A sequence of the [Apk] files which are being refreshed.
+     */
+    internal fun write(options: PatcherOptions, folder: File) = all.map {
+        val file = folder.resolve(it.toString())
+        var exception: Apk.ApkException? = null
+        try {
+            it.write(options, file)
+        } catch (e: Apk.ApkException) {
+            exception = e
+        }
+
+        SplitApkResult(it, file, exception)
+    }
+
+    inner class GlobalResources {
```

Comment:
Why is this class part of an ApkBundle and what is it used for? Please document it.